### ISSUE TITLE
When partition transitioning is stuck it is stuck for ever resulting in unusable partition

### ISSUE THREAD

**Issue by deepthidevaki** [camunda/zeebe#12763]
**Describe the bug**

In a support case we observed the following: 
ZeebePartition is transitioning to a new role. Before that it closes services from the previous role. For some reason, the close was stuck. In this case it was ExporterDirector. From the logs it looks like the ExporterDirector close was not completed. ZeebePartition is now waiting for this future for ever. Since this step is not completed, all the following transition were queued up in ZeebePartition. As a result this partition was not doing anything useful. Sometimes, this node became the leader. Since the ZeebePartition could not transition to leader no processing services where installed, resulting in this partition being effectively unavailable.

We do not have enough data such as heap dump to investigate why the close was never completed. But we should improve handling of such cases independent of the root cause. When such cases happen, there should be a way to detect it and force close the actor so that the following transitions can make progress and install necessary services.

**Solution**
We could do the following:
- Detect partition transition is stuck. It could be based on a timeout. For example, if the transition is not completed with in 60 seconds, this triggers something to notify the operator. To start with, it could simply be an error logged to the console.
- When this is detected, force close the actor which is stuck, close current transition and continue with the next transitions if anything is queued. It is important to force close the actor because if for any reason the actor continues to work this can lead to inconsistency.

**Work around:**
Restart affected broker. But, it is not easy for a user to recognize this case. It requires internal knowledge of Zeebe to recognize this scenario after analyzing the logs thoroughly.

**Environment:**
- Zeebe Version: 8.1.9 (in the support ticket). But this might affect multiple versions.

Related to https://jira.camunda.com/browse/SUPPORT-16924

**Comment by megglos** [comment]
Triage:
- came up due to an issue in a custom exporter
- minimum-scope: at least log but not continue anything
- right now not a priority will put it into the backlog for later

**Comment by oleschoenburg** [comment]
Related to #13042, we should detect this as an unhealthy partition.

Also related to #11849, timers should not be blocked by a single blocking actor job.
